doc/configuration.rst

```diff
@@ -79,27 +85,27 @@ supported by PSyclone.
 
 .. tabularcolumns:: |l|L|
 
-=======================	=======================================================
+======================= =======================================================
 Entry         		Description
-=======================	=======================================================
-SUPPORTEDAPIS 		A comma-separated list of the names of the various APIs
-                        that PSyclone supports.
+======================= =======================================================
 DEFAULTAPI              The API that PSyclone assumes an Algorithm/Kernl
                         conforms to if no API is specified. Must be one of the
-			SUPPORTEDAPIS.
-SUPPORTEDSTUBAPIS       Comma-separated list of the APIs that the kernel-stub
-                        generator (see :ref:`stub-generation`) supports.
-DEFAULTSTUBAPI          The API that the kernel-stub generator assumes by
+                        APIs supported by PSyclone. If there is no DEFAULTAPI
+                        specified but only one API specific section is in the
+                        config file loaded, this API will be used as
                         default.
+DEFAULTSTUBAPI          The API that the kernel-stub generator assumes by
+                        default. Must be one of the stub-APIs supported by
+                        PSyclone.
 DISTRIBUTED_MEMORY      Whether or not to generate code for distributed-memory
                         parallelism by default.  Note that this is currently
-			only supported for the dynamo0.3 API.
+                        only supported for the dynamo0.3 API.
 REPRODUCIBLE_REDUCTIONS Whether or not to generate code for reproducible OpenMP
                         reductions (see :ref:`openmp-reductions`) by default.
 REPROD_PAD_SIZE         If generating code for reproducible OpenMP reductions,
                         this setting controls the amount of padding used
-			between elements of the array in which each thread
-			accumulates its local reduction. (This prevents false
+                        between elements of the array in which each thread
+                        accumulates its local reduction. (This prevents false
                         sharing of cache lines by different threads.)
 ======================= =======================================================
 
```

Comment:
One suggestion that might make the situation a bit more convenient: what about instead of putting DEFAULTAPI into the config file, we only put API in the config file. Then we can say that we have a default, can print it in the help messages, and still maintain the functionality we have (with this patch) ... as far as I can see.

Reply:
Oh, I like that. So we put a default in e.g. `psyGen.py` but overwrite that if API is in the config file? If that's what you're proposing then I'm happy with that.

Reply:
Mostly, except I would put the API default in Config._default_api (so that all configuration items are in one place), and then in psyGen.get_api() use:
```
    if api == "":
        api = Config.get().api    # instead of api = Config.get().default_api
```
If the user does not set an API (neither as parameter to psyGen.get_api(), nor on the command line or config file), then the Config._api will be set to Config._default_api, so get_api() will then return the default_api.

Reply:
Oh yes - ignore me! I was forgetting we have a Config class for that now :-)

Reply:
OK, that should all be done now. One very minor question: some classes take an api as an optional argument (psyGen.get_api, parse.KernelTypeFactory.__init__, parse.parse), and if none is specified, they use Config.get().default_api. It might make sense to use Config.get().api (which is the same as default_api if no api is specified on the command line or a config file anyway) instead ... but I don't think it actually matters :)